### PR TITLE
Custom column sizing for tableView

### PR DESCRIPTION
**Pull request by sarthak-marwaha**

### CODE REVIEW

File: src/foam/u2/table/TableHeaderComponent.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  imports: [
+    'colWidthUpdated',
+    'props',
+    'selectedColumnsWidth'
+  ],
+
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'showResize'
```
Comment (kgrgreer) [inline src/foam/u2/table/TableHeaderComponent.js:25]: Should this default to true?

File: src/foam/u2/table/TableHeaderComponent.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    'oldX_',
+    'oldCW_',
+    ['isDragging_', false],
+  ],
+
+  methods: [
+    function render() {
+      var self = this;
+      var view = this.data;
+      this.propName = this.columnHandler.checkIfArrayAndReturnPropertyNamesForColumn(this.col);
```
Comment (kgrgreer) [inline src/foam/u2/table/TableHeaderComponent.js]: I'm stillNotKeenOnSuperLongButDescriptiveMethodNames.

File: src/foam/u2/table/UnstyledTableRow.js [hunk 3]
```diff
@@ -167,3 +155,57 @@ foam.CLASS({
...
+  ],
+
+  properties: [
+    {
+      name: 'colWidth',
+      factory: function() {
+        return this.selectedColumnsWidth && this.selectedColumnsWidth[this.propName] ?
+        this.selectedColumnsWidth[this.propName] :
+        this.columnHandler.returnPropertyForColumn(this.props, this.data.of, this.col, 'tableWidth');
+      },
```
Comment (kgrgreer) [inline src/foam/u2/table/UnstyledTableRow.js]: no trailing commas

File: src/foam/u2/table/TableHeaderComponent.js (hunk 1 above)
Comment (sarthak-marwaha) [reply src/foam/u2/table/TableHeaderComponent.js:25]: This controls visibility of the drag icon. which only appears on hover in order to provide as much space to the column header as possible so it defaults to false.

**Pull request merged**